src/openrct2/actions/LandSetRightsAction.hpp

```diff
@@ -0,0 +1,231 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../Context.h"
+#include "../OpenRCT2.h"
+#include "../actions/LandSetHeightAction.hpp"
+#include "../audio/audio.h"
+#include "../interface/Window.h"
+#include "../localisation/Localisation.h"
+#include "../localisation/StringIds.h"
+#include "../management/Finance.h"
+#include "../ride/RideData.h"
+#include "../windows/Intent.h"
+#include "../world/Park.h"
+#include "../world/Scenery.h"
+#include "../world/Sprite.h"
+#include "../world/Surface.h"
+#include "GameAction.h"
+
+enum class LandSetRightSetting : uint8_t
+{
+    UnownLand,
+    UnownConstructionRights,
+    SetForSale,
+    SetConstructionRightsForSale,
+    SetOwnershipWithChecks,
+    Count
+};
+
+DEFINE_GAME_ACTION(LandSetRightsAction, GAME_COMMAND_SET_LAND_OWNERSHIP, GameActionResult)
+{
+private:
+    MapRange _range;
+    uint8_t _setting = static_cast<uint8_t>(LandSetRightSetting::Count);
+    uint8_t _ownership;
```

Comment:
Should have a default invalid value and tested in Query

Reply:
Its default `0` from the two main constructors. I don't think it makes sense to default invalid this one as there isn't really an invalid value.

Reply:
When it comes from the network it will use the default constructor where this value is uninitialized, this is just to catch serialization issues.

Reply:
Okay I will default construct it to `0`. I can't do a check on that though.

Reply:
Thats why its best to have it to some invalid value like 0xFF which can be checked in Query, but its okay for now, I'll run the fuzzer soon and see what kind of issues popup when network receives garbage/missing data.

Reply:
I kind of want to just get rid of the whole `SetOwnershipWithChecks` which would make this field pointless as its too powerful a command. Its a job for another day though. 0xFF would also be a valid value thats the problem. Its a collection of flags and it will do sensible things with any value.

Reply:
Alrite, not gonna block this PR any longer and my lazyness already stalled it for too long.